src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -446,5 +425,76 @@ private void OnSecretsChanged(object sender, SecretsChangedEventArgs e)
 
             await _repository.PurgeOldSecretsAsync(currentFunctions, traceWriter, logger);
         }
+
+        private async Task<HostSecrets> LoadHostSecretsWithRetryAsync()
+        {
+            int currentRetry = 0;
+            for (; ;)
+            {
+                try
+                {
+                    HostSecrets hostSecrets = await LoadSecretsAsync<HostSecrets>();
+                    if (hostSecrets == null)
+                    {
+                        _traceWriter.Verbose(Resources.TraceHostSecretGeneration);
+                        _logger?.LogDebug(Resources.TraceHostSecretGeneration);
+                        hostSecrets = GenerateHostSecrets();
+                        await PersistSecretsAsync(hostSecrets);
+                    }
+                    return hostSecrets;
+                }
+                catch (Exception ex)
```

Comment:
If we switch this to catch the more specific `IOException`, we'll only fall into our catch if we match the exception type and avoid doing so when it isn't. That also allows us to remove the `IsTransient` method as it is the only thing it's checking for.

You can also use an exception filter, which would improve this (using the `when` clause)